syntax = "proto3";

package proto;

import "proto/types.proto";

service Indexer {
    rpc IndexPiece (IndexRequest) returns (IndexResponse);
}

enum IndexType {
    UNKNOWN = 0; // forward compatibility with enum version change
    NOTE = 1;
    MEASURE = 2;
    VECTORS = 3;
}

message IndexRequest {
    Piece piece = 1;
    IndexType type = 2;
}

message IndexResponse {
    repeated Note notes = 1;
    repeated Measure measures = 2;
    repeated Vector vectors = 3;
}
